airflow-core/src/airflow/models/taskmap.py

```diff
@@ -161,7 +161,7 @@ def expand_mapped_task(
                 )
             total_length = None
 
-        state: TaskInstanceState | None = None
+        state: TaskInstanceState | str | None = None
```

Comment:
This is only assigned once on line 217, always str | None, so the TaskInstanceState case is not needed.


```suggestion
        state: str | None = None
```

Reply:
Good catch!